R/zchunk_L2321.cement_CHINA.R

```diff
@@ -0,0 +1,549 @@
+      L2321.StubTechCoef_cement_CHINA
+
+
+    # Create the calibrated input of fuel consumption for producing heat input table.
+    #
+    # Start by subsetting the energy inputs to cement production by province for model base years,
+    # rounding to the correct xml digit and adding a region column.
+    L1321.in_EJ_province_cement_F_Y %>%
+      filter(year %in% MODEL_BASE_YEARS) %>%
+      mutate(calibrated.value = signif(value, gcamchina.DIGITS_CALOUTPUT), region = province) %>%
+      select(-value)  ->
+      L2321.StubTechCalInput_cement_heat_CHINA
+
+    # Add supplysector / subsector / technology / minicam.energy.input information
+    # from the calibrated technology mapping file to the energy inputs to cement production
+    # data frame.
+    L2321.StubTechCalInput_cement_heat_CHINA %>%
+      left_join_error_no_match(calibrated_techs %>%
+                                 select(sector, fuel, supplysector, subsector, technology, minicam.energy.input),
+                               by = c("sector", "fuel")) ->
+      L2321.StubTechCalInput_cement_heat_CHINA
+
+    # Since this table should only contain the technologies for producing heat, remove
+    # electricity inputs to the cement production technology from the stub technology coefficients by
+    # filtering for supply sectors NOT included in the L2321.StubTechCoef_cement_CHINA data frame which
+    # contains electricity inputs.
+    L2321.StubTechCalInput_cement_heat_CHINA %>%
+      filter(!supplysector %in% L2321.StubTechCoef_cement_CHINA$supplysector) ->
+      L2321.StubTechCalInput_cement_heat_CHINA_NOelectricity
+
+    # Add region and share weight information, format.
+    L2321.StubTechCalInput_cement_heat_CHINA_NOelectricity %>%
+      mutate(region = province,
+             stub.technology = technology,
+             share.weight.year = year,
+             subs.share.weight = if_else(calibrated.value > 0, 1, 0),
+             tech.share.weight = subs.share.weight) %>%
+      select(region, supplysector, subsector, stub.technology, year, minicam.energy.input,
+             calibrated.value, share.weight.year, subs.share.weight, tech.share.weight) ->
+      L2321.StubTechCalInput_cement_heat_CHINA
+
+
+    # Create the input table with the market names  for the fuels consumed for heat by the province cement sectors
+    #
+    # Add model years to the stub technology coefficients for the cement sector. Then remove the
+    # cement supplysector leaving only the process heat supplysector.
+    L2321.StubTech_cement_CHINA %>%
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      filter(!supplysector %in% L2321.StubTechCoef_cement_CHINA$supplysector) ->
+      L2321.StubTechMarket_cement_CHINA
+
+    # Use the calibrated technology mapping data frame to add minicam.energy.input information to
+    # the process heat supplysector data frame.
+    L2321.StubTechMarket_cement_CHINA %>%
+      # Use left join here to pass time shift test.
```

Comment:
Reviewers might ask question about this. Are there NAs that need to be preserved? What happens to the data table when we use let join vs left join error no match